drivers/platform/stm32/stm32_gpio_irq.h

```diff
@@ -0,0 +1,70 @@
+/***************************************************************************//**
+ *   @file   stm32/stm32_gpio_irq.h
+ *   @brief  Header file for stm32 gpio irq specifics.
+ *   @author Ramona Bolboaca (ramona.bolboaca@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef STM32_GPIO_IRQ_H
+#define STM32_GPIO_IRQ_H
+
+#include "no_os_irq.h"
+#include "stm32_hal.h"
+
+/**
+ * @struct stm32_gpio_irq_init_param
+ * @brief Structure holding the initialization parameters for stm32 platform
+ * specific GPIO IRQ parameters.
+ */
+struct stm32_gpio_irq_init_param {
+	uint8_t port_nb;
+};
+
+/**
+ * @struct stm32_gpio_irq_desc
+ * @brief stm32 platform specific GPIO IRQ descriptor
+ */
+struct stm32_gpio_irq_desc {
```

Comment:
this should probably also have a pointer to its parent, no_os_irq_desc

Reply:
seeing how we always use the HAL APIs to handle gpio irq controller, I don't really see any usage for a parent reference.